### DISCUSSION TITLE
include width in generating itinerary for bikes

### DISCUSSION THREAD

**Discussion by wortan666** [osmandapp/OsmAnd#19648] [Ideas]
Hello,
I've been discovering this app, and I love the fact that it uses a lot of openstreetmap data. 
So I've got a suggestion that would make its use even better. 
As you probably know, more and more people choose to travel by bicycle, whether it is for environmental issues, health benefits, or cost. 
Some choose to travel light, others want to bring a bit more stuff, using small or less small bike trailers.
And that's where there could be some improvement. These bike trailers, whether it's small trailer to bring some stuff, or bigger trailer, like bike campers, have various width, and using paths or cycle paths can be tricky, cause most have very variable width limitations, from 60 cm to 2m, and carrying a heavy 1.2m wide bike camper can be frustrating when you have a 90cm width limitation and you have to go back and find another way. The problem is that no application takes that into account.
But, and that's why I'm turning to you, Openstreetmap does have the possibility to enter as a parameter of each path section (cycle or classic) the width of the path. So if it would be possible to add that parameter to your app, it would be possible for someone having a 1m wide trailer for exemple, to add that before searching for the best path, and the app would automatically rule out any path with a registered width under 1m... of course if he encounters some narrower paths that have not been registered yet, he can participate and registered them for the future.

I hope everything is clear (I'm not a native english speaker). I'm sure it would benefit a lot of people, as this option doesn't exist yet and more and more people want to avoid the main roads while cycling while carrying different kinds of bike trailers.

**Comment by DmitryAlexei** [comment]
@wortan666 please provide further information about tags, that OsmAnd needs to support. 
I searched on the OpenStreetMap wiki and found [cycling_width](https://taginfo.openstreetmap.org/keys/cycling_width#values) tag. Although the values of this tag are mostly not suitable for routing (they are not numeric values).
